src/script/lua_api/l_client.cpp

```diff
@@ -116,6 +116,11 @@ int ModApiClient::l_clear_out_chat_queue(lua_State *L)
 // get_player_names()
 int ModApiClient::l_get_player_names(lua_State *L)
 {
+	if (getClient(L)->checkCSMRestrictionFlag(
+			CSMRestrictionFlags::CSM_RF_READ_PLAYERINFO)) {
+		return 0;
```

Comment:
Why not just return an empty table?

Reply:
this permits mods to know if it's disabled or if there is no player